internal/services/api/post_connect_cloud_device_auth.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+// Copyright (C) 2025 by Posit Software, PBC.
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/posit-dev/publisher/internal/clients/cloud_auth"
+	"net/http"
+	"time"
+
+	"github.com/posit-dev/publisher/internal/logging"
+)
+
+var cloudAuthClientFactory = cloud_auth.NewCloudAuthClient
+
+const cloudAuthBaseURLHeader = "Cloud-Auth-Base-Url"
+
+type connectCloudDeviceAuthResponseBody struct {
+	DeviceCode              string `json:"deviceCode"`
+	UserCode                string `json:"userCode"`
+	VerificationURIComplete string `json:"verificationURIComplete"`
+	Interval                int    `json:"interval"`
```

Comment:
For `json` notation, when possible, we stick to underscore syntax `json:"deviceCode"` -> `json:"device_code"`

Reply:
It looks like the convention for the agent API is camel case. I see `caCert`, `configurationName`, `preferredPythonPath`, etc.